### PR TITLE
[RFR] WithPermission component

### PR DESCRIPTION
**Pull request by djhi**
Follow #1 

A simpler component which will render its children only if its permissions are matched

### CODE REVIEW

File: src/SwitchPermissions.js [hunk 1]
```diff
@@ -56,6 +58,6 @@ export default class SwitchPermissions extends Component {
             return createElement(loading);
         }
 
-        return cloneElement(match, props);
+        return <div>{React.Children.map(match, child => <FormField input={cloneElement(child)} {...props} />)}</div>;
```
Comment (djhi) [inline src/SwitchPermissions.js]: Not sure the `map` is a good idea here as it implies wrapping the children inside a parent element (`div`)

File: src/WithPermission.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            }
+
+            return null;
+        }
+
+        if (!match) {
+            return createElement(loading);
+        }
+
+        return <div>{React.Children.map(match, child => <FormField input={cloneElement(child)} {...props} />)}</div>;
```
Comment (djhi) [inline src/WithPermission.js]: Not sure the `map` is a good idea here as it implies wrapping the children inside a parent element (`div`)

File: src/SwitchPermissions.js (hunk 1 above)
Comment (djhi) [reply src/SwitchPermissions.js]: This might have consequences on styles

Comment (fzaninotto) [reply src/SwitchPermissions.js]: then make it a `span`

File: src/WithPermission.js (hunk 2 above)
Comment (fzaninotto) [reply src/WithPermission.js]: why do you `clone` the child here? `FormField` will clone it anyway.

**Review by fzaninotto — CHANGES_REQUESTED** [review]

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**